### ISSUE TITLE
10.0.3 — Uncaught TypeError: Cannot read property 'push' of undefined

### ISSUE THREAD

**Issue by pnfcre** [preactjs/preact#2059]
Hi there,

My project stopped working when upgrading to Preact 10.0.3 a few minutes ago. This change was later reverted, but you can find the source code in its current state here: https://github.com/next/zeiw-client. If you want to reproduce the error, feel free to clone that repository and upgrade Preact to 10.0.3 👍

![image](https://user-images.githubusercontent.com/24176136/67766042-d84f1d00-fa4d-11e9-96e4-0632fc7b640c.png)

And here's the failing `preact.config.js` causing the error:

```js
import child_process from 'child_process'
import webpack from 'webpack'

const commitHash = child_process
  .execSync('git rev-parse HEAD')
  .toString()
  .trim()

export default ({ plugins }) => {
  plugins.push(
    new webpack.DefinePlugin({
      '_zeiwBuild.commitHash': JSON.stringify(commitHash)
    })
  )
}
```

Kind regards,
Hampus Kraft.

**Comment by Raul-MultiSafepay** [comment]
I have just upgrade as well and i get another error on the hooks file
`hooks.module.js:1 Uncaught TypeError: Cannot read property 'forEach' of undefined
    at hooks.module.js:1
    at Array.some (<anonymous>)
    at Object../node_modules/preact/hooks/dist/hooks.module.js.preact__WEBPACK_IMPORTED_MODULE_0__.options.__c (hooks.module.js:1)
    at $ (preact.module.js:1)
    at H (preact.module.js:1)`

I didn't create a new issue as maybe could be related because is the same file

**Comment by mhmdanas** [comment]
I'm having the same issue, but instead of `push` it's `forEach`.

Here's the stacktrace:

<img width="708" alt="Screen Shot 2019-10-29 at 3 39 38 PM" src="https://user-images.githubusercontent.com/32234660/67767825-53223300-fa62-11e9-9e31-73dcee1b9c66.png">

**Comment by JoviDeCroock** [comment]
There's a PR with the fix and a patch release will soon be there :)

**Comment by robertknight** [comment]
This issue has been resolved in [v10.0.4](https://github.com/preactjs/preact/releases/tag/10.0.4). We'll look into options to prevent a future reoccurrences.
